scale-tests/streaming_test.py

```diff
@@ -0,0 +1,101 @@
+import json
+import sys
+
+import sdk_cmd
+import shakedown
+import spark_utils as utils
+from tests import test_kafka
+
+KAFKA_PACKAGE_NAME = "kafka"
+KAFKA_SERVICE_NAME = "kafka"
+PRODUCER_SERVICE_NAME = "Spark->Kafka Producer"
+
+
+def run_pipeline(kerberos_flag, num_consumers, stop_count, jar_uri, spark_app_name):
+    stop_count = str(stop_count)
+    broker_dns = _kafka_broker_dns()
+    topic = "top1"
+
+    big_file, big_file_url = "file:///mnt/mesos/sandbox/big.txt", "http://norvig.com/big.txt"
+
+    # arguments to the application
+    producer_args = " ".join([broker_dns, big_file, topic, kerberos_flag])
+
+    uris = "spark.mesos.uris=http://norvig.com/big.txt"
+
+    common_args = [
+        "--conf", "spark.mesos.containerizer=mesos",
+        "--conf", "spark.scheduler.maxRegisteredResourcesWaitingTime=2400s",
+        "--conf", "spark.scheduler.minRegisteredResourcesRatio=1.0",
+        "--conf", uris
+    ]
+
+    producer_config = ["--conf", "spark.cores.max=2", "--conf", "spark.executor.cores=2",
+                       "--class", "KafkaFeeder"] + common_args
+
+    producer_id = utils.submit_job(app_url=jar_uri,
+                                   app_args=producer_args,
+                                   app_name=spark_app_name,
+                                   args=producer_config,
+                                   verbose=False)
+
+    shakedown.wait_for(lambda: test_kafka._producer_launched(), ignore_exceptions=False, timeout_seconds=600)
```

Comment:
If the timeout is reached, this will throw an exception and the script will fail. I think we can omit this line ... just submit and continue. Then if the job gets stuck in the Dispatcher queue because of some bug, we can measure how long it's stuck in the metrics we're monitoring (that Nick is adding) and which may be longer than 600 seconds.

Reply:
okay I'll omit